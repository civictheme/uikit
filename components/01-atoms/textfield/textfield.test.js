const template = 'components/01-atoms/textfield/textfield.twig';

describe('Textfield Component', () => {
  test('renders with required attributes', async () => {
    const c = await dom(template, {
      name: 'test-textfield',
      value: 'Sample text',
    });

    expect(c.querySelectorAll('.ct-textfield')).toHaveLength(1);
    expect(c.querySelector('.ct-textfield').getAttribute('value')).toEqual('Sample text');
    expect(c.querySelector('.ct-textfield').getAttribute('name')).toEqual('test-textfield');

    assertUniqueCssClasses(c);
  });

  test('renders with optional attributes', async () => {
    const c = await dom(template, {
      name: 'test-textfield',
      value: 'Sample text',
      placeholder: 'Enter text here',
      is_invalid: true,
      is_disabled: true,
      is_required: true,
      attributes: 'data-test="true"',
      modifier_class: 'custom-class',
      theme: 'dark',
    });

    expect(c.querySelectorAll('.ct-textfield.custom-class.ct-theme-dark.ct-textfield--is-invalid')).toHaveLength(1);
    expect(c.querySelector('.ct-textfield').getAttribute('name')).toEqual('test-textfield');
    expect(c.querySelector('.ct-textfield').getAttribute('value')).toEqual('Sample text');
    expect(c.querySelector('.ct-textfield').getAttribute('placeholder')).toEqual('Enter text here');
    expect(c.querySelector('.ct-textfield').getAttribute('disabled')).toBe('');
    expect(c.querySelector('.ct-textfield').getAttribute('required')).toBe('');
    expect(c.querySelector('.ct-textfield').getAttribute('data-test')).toEqual('true');

    assertUniqueCssClasses(c);
  });

  test('does not render when name is empty', async () => {
    const c = await dom(template, {
      name: '',
    });

    expect(c.querySelectorAll('.ct-textfield')).toHaveLength(0);
  });

  test('strips HTML tags from attribute values', async () => {
    const c = await dom(template, {
      name: 'test-textfield',
      value: 'Sample text',
      attributes: 'data-test="<script>alert(1)</script>"',
    });

    expect(c.querySelector('.ct-textfield').getAttribute('value')).toEqual('Sample text');
    expect(c.querySelector('.ct-textfield').getAttribute('name')).toEqual('test-textfield');
    expect(c.querySelector('.ct-textfield').getAttribute('data-test')).toEqual('alert(1)');

    assertUniqueCssClasses(c);
  });
});
