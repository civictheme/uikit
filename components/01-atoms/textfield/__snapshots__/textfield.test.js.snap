// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textfield Component does not render when name is empty 1`] = `
<div>
  



</div>
`;

exports[`Textfield Component renders with optional attributes 1`] = `
<div>
  


	
  <input
    class="ct-textfield ct-theme-dark ct-textfield--is-invalid custom-class"
    data-test="true"
    disabled=""
    name="test-textfield"
    placeholder="Enter text here"
    required=""
    type="textfield"
    value="Sample text"
  />
  

</div>
`;

exports[`Textfield Component renders with required attributes 1`] = `
<div>
  


	
  <input
    class="ct-textfield ct-theme-light"
    name="test-textfield"
    type="textfield"
    value="Sample text"
  />
  

</div>
`;

exports[`Textfield Component strips HTML tags from attribute values 1`] = `
<div>
  


	
  <input
    class="ct-textfield ct-theme-light"
    data-test="<script>alert(1)</script>"
    name="test-textfield"
    type="textfield"
    value="Sample text"
  />
  

</div>
`;
